phoenicis-apps/src/main/java/org/phoenicis/apps/GitApplicationsSource.java

```diff
@@ -18,45 +18,109 @@
+import org.phoenicis.apps.dto.CategoryDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 class GitApplicationsSource implements ApplicationsSource {
-    private final String gitRepositoryURL;
-    private final LocalApplicationsSource.Factory localAppsSourceFactory;
-    private List<CategoryDTO> cache;
-
-    GitApplicationsSource(String gitRepositoryURL, LocalApplicationsSource.Factory localAppsSourceFactory) {
-        this.gitRepositoryURL = gitRepositoryURL;
-        this.localAppsSourceFactory = localAppsSourceFactory;
-    }
-
-    @Override
-    public synchronized List<CategoryDTO> fetchInstallableApplications() {
-        if (cache != null) {
-            return cache;
-        }
-
-        try {
-            final File gitTmp = Files.createTempDirectory("git").toFile();
-            gitTmp.deleteOnExit();
-            new ProcessBuilder(Arrays.asList("git", "clone", gitRepositoryURL, gitTmp.getAbsolutePath()))
-                    .inheritIO()
-                    .start()
-                    .waitFor();
-
-            cache = localAppsSourceFactory.createInstance(gitTmp.getAbsolutePath()).fetchInstallableApplications();
-            return cache;
-        } catch (IOException | InterruptedException e) {
-            return Collections.emptyList();
-        }
-    }
+	private final static Logger LOGGER = LoggerFactory.getLogger(GitApplicationsSource.class);
+
+	private final String gitRepositoryURL;
+	private final LocalApplicationsSource.Factory localAppsSourceFactory;
+	private List<CategoryDTO> cache;
+
+	private static final String cacheDirectoryPath = System.getProperty("java.io.tmpdir");
+
+	GitApplicationsSource(String gitRepositoryURL, LocalApplicationsSource.Factory localAppsSourceFactory) {
+		this.gitRepositoryURL = gitRepositoryURL;
+		this.localAppsSourceFactory = localAppsSourceFactory;
+	}
+
+	@Override
+	public synchronized List<CategoryDTO> fetchInstallableApplications() {
+		if (cache != null) {
+			return cache;
+		}
+
+		File gitRepositoryLocation = new File(
+				GitApplicationsSource.cacheDirectoryPath + "/git" + this.gitRepositoryURL.hashCode());
+
+		LOGGER.info(String.format("Git-repository '%s' will be saved in '%s'", this.gitRepositoryURL,
+				gitRepositoryLocation.getAbsolutePath()));
+
+		/*
+		 * This value is true if a folder for the git repository exists,
+		 * otherwise it is false
+		 */
+		boolean folderExists = true;
+
+		// check that the repository folder exists
+		if (!gitRepositoryLocation.exists()) {
+			folderExists = false;
+
+			LOGGER.info(String.format("Creating new folder '%s' for git-repository '%s'",
+					gitRepositoryLocation.getAbsolutePath(), this.gitRepositoryURL));
+
+			if (!gitRepositoryLocation.mkdir()) {
+				LOGGER.error(String.format("Couldn't create folder for git repository '%s' at '%s'",
+						this.gitRepositoryURL, gitRepositoryLocation.getAbsolutePath()));
+
+				return Collections.emptyList();
```

Comment:
Don't think that's a good idea. Assume you want to start an already installed app but cannot because the apps cannot be fetched. Also, you could have many other ApplicationsSources and not even care about the git one.